### PR TITLE
proxy: implement compaction function

### PR DESCRIPTION
**Pull request by mqliang**
implement compaction function in proxy.

@xiang90

### CODE REVIEW

File: proxy/grpcproxy/kv.go [hunk 1]
```diff
@@ -138,6 +149,11 @@ func DelRequestToOp(r *pb.DeleteRangeRequest) clientv3.Op {
 	return clientv3.OpDelete(string(r.Key), opts...)
 }
 
-func (p *kvProxy) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.CompactionResponse, error) {
-	panic("unimplemented")
+func CompactionRequestToOp(r *pb.CompactionRequest) clientv3.Op {
+	opts := []clientv3.OpOption{}
```
Comment (gyuho) [inline proxy/grpcproxy/kv.go]: nit. `var opts []clientv3.OpOption` so it can be `nil` when not physical?

File: proxy/grpcproxy/kv.go [hunk 2]
```diff
@@ -81,8 +81,17 @@ func (p *kvProxy) Txn(ctx context.Context, r *pb.TxnRequest) (*pb.TxnResponse, e
 	return (*pb.TxnResponse)(resp), err
 }
 
+func (p *kvProxy) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.CompactionResponse, error) {
+	resp, err := p.client.Do(ctx, CompactionRequestToOp(r))
```
Comment (xiang90) [inline proxy/grpcproxy/kv.go]: why not just use https://github.com/coreos/etcd/blob/master/clientv3/kv.go#L101?

Comment (xiang90) [inline proxy/grpcproxy/kv.go]: i do not think we should introduce compact op.

Comment (mqliang) [inline proxy/grpcproxy/kv.go]: because  https://github.com/coreos/etcd/blob/master/clientv3/kv.go#L101 can just specify revision, can not set physical

Comment (xiang90) [inline proxy/grpcproxy/kv.go]: we can extend that interface to make it accept options.

Comment (mqliang) [inline proxy/grpcproxy/kv.go]: Ok, delegate compaction request to client directly SGTM.

**Comment by mqliang** [comment]
@xiang90 @gyuho Updated. PTAL.

**Comment by gyuho** [comment]
lgtm. Defer to @xiang90 @heyitsanthony 

Thanks.

**Comment by heyitsanthony** [comment]
I thought @xiang90 meant something like `Compact(ctx context.Context, rev int64, opts CompactOptions...)` instead of `Compact(ctx context.Context, rev int64, physical bool)`?

**Comment by gyuho** [comment]
Yeah passing a typed option arg sounds better. Something like `WithPhysical`?

**Comment by xiang90** [comment]
@heyitsanthony Right.

**Comment by xiang90** [comment]
let's get this rebased with master and merged.

**Comment by xiang90** [comment]
lgtm. I am going to write code for proxy setup. i will ping you for a review when it is ready.

**Comment by mqliang** [comment]
cc/ @jerryzh168

**Pull request merged**